engine/apps/public_api/serializers/incidents.py

```diff
@@ -50,7 +40,7 @@ def get_title(self, obj):
         return obj.web_title_cache
 
     def get_alerts_count(self, obj):
-        return len(obj.alerts.all())
+        return obj.alerts.count()
```

Comment:
This count query is more efficient than prefetching alerts (loading them all in mem) and count them later.